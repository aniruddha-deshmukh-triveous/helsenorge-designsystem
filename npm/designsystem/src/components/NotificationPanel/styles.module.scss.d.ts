export const anchorlink: string
export const anchorlinkSpacer: string
export const anchorlink__spacer: string
export const anchorlinkIconWrapper: string
export const anchorlinkIcon: string
export const anchorlink__icon: string
export const notificationPanel: string
export const notificationPanelSmall: string
export const notificationPanelMedium: string
export const notificationPanelLarge: string
export const notificationPanelInfo: string
export const notificationPanelWarn: string
export const notificationPanelAlert: string
export const notificationPanelCrisis: string
export const notificationPanelShadow: string
export const notificationPanelHaslabel: string
export const notificationPanelFluidwrapper: string
export const notificationPanelFluidwrapperInfo: string
export const notificationPanelFluidwrapperWarn: string
export const notificationPanelFluidwrapperAlert: string
export const notificationPanelFluidwrapperCrisis: string
export const notificationPanelFluidwrapperShadow: string
export const notificationPanelIcon: string
export const notificationPanelContent: string
export const notificationPanelContentCrisis: string
export const notificationPanelContentIsred: string
export const notificationPanelContentHaslabel: string
export const notificationPanelLabel: string
export const notificationPanelActionColumn: string
export const notificationPanelCloseButton: string
interface Namespace {
	"anchorlink": string,
	"anchorlinkSpacer": string,
	"anchorlink__spacer": string,
	"anchorlinkIconWrapper": string,
	"anchorlink__icon-wrapper": string,
	"anchorlinkIcon": string,
	"anchorlink__icon": string,
	"notificationPanel": string,
	"notification-panel": string,
	"notificationPanelSmall": string,
	"notification-panel--small": string,
	"notificationPanelMedium": string,
	"notification-panel--medium": string,
	"notificationPanelLarge": string,
	"notification-panel--large": string,
	"notificationPanelInfo": string,
	"notification-panel--info": string,
	"notificationPanelWarn": string,
	"notification-panel--warn": string,
	"notificationPanelAlert": string,
	"notification-panel--alert": string,
	"notificationPanelCrisis": string,
	"notification-panel--crisis": string,
	"notificationPanelShadow": string,
	"notification-panel--shadow": string,
	"notificationPanelHaslabel": string,
	"notification-panel--haslabel": string,
	"notificationPanelFluidwrapper": string,
	"notification-panel__fluidwrapper": string,
	"notificationPanelFluidwrapperInfo": string,
	"notification-panel__fluidwrapper--info": string,
	"notificationPanelFluidwrapperWarn": string,
	"notification-panel__fluidwrapper--warn": string,
	"notificationPanelFluidwrapperAlert": string,
	"notification-panel__fluidwrapper--alert": string,
	"notificationPanelFluidwrapperCrisis": string,
	"notification-panel__fluidwrapper--crisis": string,
	"notificationPanelFluidwrapperShadow": string,
	"notification-panel__fluidwrapper--shadow": string,
	"notificationPanelIcon": string,
	"notification-panel__icon": string,
	"notificationPanelContent": string,
	"notification-panel__content": string,
	"notificationPanelContentCrisis": string,
	"notification-panel__content--crisis": string,
	"notificationPanelContentIsred": string,
	"notification-panel__content--isred": string,
	"notificationPanelContentHaslabel": string,
	"notification-panel__content--haslabel": string,
	"notificationPanelLabel": string,
	"notification-panel__label": string,
	"notificationPanelActionColumn": string,
	"notification-panel__action-column": string,
	"notificationPanelCloseButton": string,
	"notification-panel__close-button": string,
}
declare const stylesModuleScss: Namespace
export default stylesModuleScss
