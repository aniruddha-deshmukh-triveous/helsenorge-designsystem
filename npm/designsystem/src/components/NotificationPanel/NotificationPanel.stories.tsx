import React from 'react';

import { ComponentStory, ComponentMeta } from '@storybook/react';

import NotificationPanel from './NotificationPanel';

export default {
  title: 'NotificationPanel',
  component: NotificationPanel,
  argTypes: {
    shadow: {
      control: 'boolean',
      defaultValue: false,
    },
    dismissable: {
      control: 'boolean',
      defaultValue: false,
    },
    size: {
      control: 'select',
      options: ['small', 'medium', 'large'],
      defaultValue: 'large',
    },
    fluid: {
      control: 'boolean',
      defaultValue: false,
    },
    label: {
      control: 'text',
      defaultValue: 'Det har skjedd noe galt. Prøv igjen senere.',
    },
    variant: {
      control: 'select',
      options: ['info', 'warn', 'alert', 'crisis'],
      defaultValue: 'alert',
    },
  },
} as ComponentMeta<typeof NotificationPanel>;

export const Default: ComponentStory<typeof NotificationPanel> = (args: any) => (
  <div style={{ width: args.fluid ? '100vw' : '1400px' }}>
    <NotificationPanel {...args}>
      {`På grunn av kommunesammenslåingen ved nyttår vil enkelte Helsenorge-tjenester være ustabile eller tidsvis utilgjenglig i en periode i begynnelsen av januar. Vi beklager ulempnene dette medfører og oppfordrer til å prøve igjen senere. `}
      <a href="/test" target="_blank" rel="noopener noreferrer">
        {'Les mer'}
      </a>
    </NotificationPanel>
  </div>
);

export const SimplifiedLabelOnly: ComponentStory<typeof NotificationPanel> = (args: any) => (
  <div style={{ width: args.fluid ? '100vw' : '80rem' }}>
    <NotificationPanel {...args} />
  </div>
);

export const AllVariants: ComponentStory<typeof NotificationPanel> = (args: any) => (
  <div className="container">
    <div className="row">
      <div className={'col-12'}>
        <NotificationPanel {...args} variant={'info'}>
          <p>
            {`På grunn av kommunesammenslåingen ved nyttår vil enkelte Helsenorge-tjenester være ustabile eller tidsvis utilgjenglig i en
            periode i begynnelsen av januar. Vi beklager ulempnene dette medfører og oppfordrer til å prøve igjen senere.`}
            <a href="/">{'Les mer om dine rettigheter her.'}</a>
          </p>
        </NotificationPanel>
      </div>
    </div>
    <div className="row mt-6">
      <div className={'col-12'}>
        <NotificationPanel {...args} variant={'warn'}>
          <p>
            {`På grunn av kommunesammenslåingen ved nyttår vil enkelte Helsenorge-tjenester være ustabile eller tidsvis utilgjenglig i en
            periode i begynnelsen av januar. Vi beklager ulempnene dette medfører og oppfordrer til å prøve igjen senere. `}
            <a href="/" target="_blank" rel="noopener noreferrer">
              {'Les mer om dine rettigheter her (external).'}
            </a>
          </p>
        </NotificationPanel>
      </div>
    </div>
    <div className="row mt-6">
      <div className={'col-12'}>
        <NotificationPanel {...args} variant={'alert'}>
          <p>
            {`På grunn av kommunesammenslåingen ved nyttår vil enkelte Helsenorge-tjenester være ustabile eller tidsvis utilgjenglig i en
            periode i begynnelsen av januar. Vi beklager ulempnene dette medfører og oppfordrer til å prøve igjen senere. `}
            <a href="/">{'Les mer om dine rettigheter her.'}</a>
          </p>
        </NotificationPanel>
      </div>
    </div>
    <div className="row mt-6">
      <div className={'col-12'}>
        <NotificationPanel {...args} variant={'crisis'}>
          <p>
            {`På grunn av kommunesammenslåingen ved nyttår vil enkelte Helsenorge-tjenester være ustabile eller tidsvis utilgjenglig i en
            periode i begynnelsen av januar. Vi beklager ulempnene dette medfører og oppfordrer til å prøve igjen senere. `}
            <a href="/" target="_blank" rel="noopener noreferrer">
              {'Les mer om dine rettigheter her.'}
            </a>
          </p>
        </NotificationPanel>
      </div>
    </div>
  </div>
);
