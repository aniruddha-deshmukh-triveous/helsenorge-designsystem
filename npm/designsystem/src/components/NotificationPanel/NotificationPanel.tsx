import React from 'react';
import cn from 'classnames';

import { palette } from '../../theme/palette';
import Icon from '../Icons';

import InfoSignStroke from '../Icons/InfoSignStroke';
import AlertSignStroke from '../Icons/AlertSignStroke';
import AlertSignFill from '../Icons/AlertSignFill';

import styles from './styles.module.scss';
import Close from '../Close';
import { AnalyticsId } from '../../constants';

export type NotificationPanelVariants = 'info' | 'warn' | 'alert' | 'crisis';
export type NotificationPanelSizes = 'small' | 'medium' | 'large';

interface NotificationPanelProps {
  /** Adds custom classes to the element. */
  className?: string;
  /** Adds inner child elements. */
  children?: React.ReactNode;
  /** Changes the visual representation of the notification panel. */
  variant?: NotificationPanelVariants;
  /** Adds a shadow effect around the notification panel. */
  shadow?: boolean;
  /** Sets a fixed size for the content container. */
  size?: NotificationPanelSizes;
  /** Used in combination with dismissiable property to close the notification panel. */
  onClick?: (e?: React.MouseEvent<HTMLElement, MouseEvent>) => void;
  /** Toggles the close button in the top right corner. Will only show if there are children. */
  dismissable?: boolean;
  /** Enables a fluid outer container that spans the entire width of parent. */
  fluid?: boolean;
  /** Sets a label for the notification panel. */
  label?: string;
  /** Close button aria-label */
  ariaLabelCloseBtn?: string;
  /** Sets the data-testid attribute. */
  testId?: string;
}

export const variantToColorMap = {
  info: { color: 'kiwi', depth: 50 },
  warn: { color: 'banana', depth: 50 },
  alert: { color: 'cherry', depth: 100 },
  crisis: { color: 'black', depth: 0 },
};

const variantToIconMap = {
  info: <Icon svgIcon={InfoSignStroke} color={palette.kiwi900} hoverColor={palette.kiwi900} />,
  warn: <Icon svgIcon={AlertSignStroke} color="black" hoverColor="black" />,
  alert: <Icon svgIcon={AlertSignFill} color={palette.cherry500} hoverColor={palette.cherry500} />,
  alertLabel: <Icon svgIcon={AlertSignStroke} color={palette.cherry500} hoverColor={palette.cherry500} />,
  crisis: <Icon svgIcon={AlertSignFill} color={palette.banana200} hoverColor={palette.banana200} />,
};

const NotificationPanel = React.forwardRef(function NotificationPanelForwardedRef(
  props: NotificationPanelProps,
  ref: React.ForwardedRef<HTMLDivElement>
) {
  const { children, variant = 'info', shadow = false, dismissable = false, onClick, label, fluid = false, size, className, testId } = props;

  function wrapFluid(panel: React.ReactElement): React.ReactElement {
    if (fluid) {
      return (
        <div
          className={cn(styles['fluid-wrapper'], styles['fluid-wrapper--' + variant], {
            [styles['fluid-wrapper--shadow']]: shadow,
          })}
        >
          {panel}
        </div>
      );
    }
    return panel;
  }

  const getStringChildren = (): string => {
    if (children) {
      let textChildren = '';

      React.Children.map(children, child => {
        if (typeof child === 'string') {
          textChildren += child;
        }
      });

      return textChildren;
    }

    return '';
  };

  const labelOnly = !!label && !children;

  return wrapFluid(
    <div
      ref={ref}
      data-testid={testId}
      data-analyticsid={AnalyticsId.NotificationPanel}
      className={cn(
        styles['notification-panel'],
        size && styles['notification-panel--' + size],
        styles['notification-panel--' + variant],
        {
          [styles['notification-panel--shadow']]: !fluid && shadow,
          [styles['notification-panel--has-children']]: !labelOnly,
          [styles['notification-panel--label-only']]: labelOnly,
          [styles['notification-panel--dismissable']]: !labelOnly && dismissable,
        },
        className
      )}
    >
      <span className={styles['notification-panel__icon']}>
        {variantToIconMap[variant === 'alert' && label && !children ? 'alertLabel' : variant]}
      </span>
      <section
        aria-label={getStringChildren()}
        className={cn(styles['notification-panel__content'], {
          [styles['notification-panel__content--crisis']]: variant === 'crisis',
          [styles['notification-panel__content--alert']]: variant === 'alert',
        })}
      >
        {label && <h1 className={styles['notification-panel__label']} dangerouslySetInnerHTML={{ __html: label }} />}
        {children}
      </section>
      {!labelOnly && dismissable && (
        <span className={styles['notification-panel__close']}>
          <Close ariaLabel={props.ariaLabelCloseBtn} onClick={onClick} />
        </span>
      )}
    </div>
  );
});

export default NotificationPanel;
