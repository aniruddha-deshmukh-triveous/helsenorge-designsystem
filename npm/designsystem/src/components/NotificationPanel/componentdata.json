{
  "props": {
    "className": {
      "defaultValue": null,
      "description": "Adds custom classes to the element.",
      "name": "className",
      "parent": {
        "fileName": "src/components/NotificationPanel/NotificationPanel.tsx",
        "name": "NotificationPanelProps"
      },
      "declarations": [
        {
          "fileName": "src/components/NotificationPanel/NotificationPanel.tsx",
          "name": "NotificationPanelProps"
        }
      ],
      "required": false,
      "type": {
        "name": "string"
      }
    },
    "children": {
      "defaultValue": null,
      "description": "Adds inner child elements.",
      "name": "children",
      "parent": {
        "fileName": "src/components/NotificationPanel/NotificationPanel.tsx",
        "name": "NotificationPanelProps"
      },
      "declarations": [
        {
          "fileName": "src/components/NotificationPanel/NotificationPanel.tsx",
          "name": "NotificationPanelProps"
        }
      ],
      "required": false,
      "type": {
        "name": "ReactNode"
      }
    },
    "variant": {
      "defaultValue": null,
      "description": "Changes the visual representation of the notification panel.",
      "name": "variant",
      "parent": {
        "fileName": "src/components/NotificationPanel/NotificationPanel.tsx",
        "name": "NotificationPanelProps"
      },
      "declarations": [
        {
          "fileName": "src/components/NotificationPanel/NotificationPanel.tsx",
          "name": "NotificationPanelProps"
        }
      ],
      "required": false,
      "type": {
        "name": "enum",
        "raw": "\"info\" | \"warn\" | \"alert\" | \"crisis\" | undefined",
        "value": [
          {
            "value": "undefined"
          },
          {
            "value": "\"info\""
          },
          {
            "value": "\"warn\""
          },
          {
            "value": "\"alert\""
          },
          {
            "value": "\"crisis\""
          }
        ]
      }
    },
    "shadow": {
      "defaultValue": null,
      "description": "Adds a shadow effect around the notification panel.",
      "name": "shadow",
      "parent": {
        "fileName": "src/components/NotificationPanel/NotificationPanel.tsx",
        "name": "NotificationPanelProps"
      },
      "declarations": [
        {
          "fileName": "src/components/NotificationPanel/NotificationPanel.tsx",
          "name": "NotificationPanelProps"
        }
      ],
      "required": false,
      "type": {
        "name": "boolean"
      }
    },
    "size": {
      "defaultValue": null,
      "description": "Sets a fixed size for the content container.",
      "name": "size",
      "parent": {
        "fileName": "src/components/NotificationPanel/NotificationPanel.tsx",
        "name": "NotificationPanelProps"
      },
      "declarations": [
        {
          "fileName": "src/components/NotificationPanel/NotificationPanel.tsx",
          "name": "NotificationPanelProps"
        }
      ],
      "required": false,
      "type": {
        "name": "enum",
        "raw": "\"small\" | \"medium\" | \"large\" | undefined",
        "value": [
          {
            "value": "undefined"
          },
          {
            "value": "\"small\""
          },
          {
            "value": "\"medium\""
          },
          {
            "value": "\"large\""
          }
        ]
      }
    },
    "onClick": {
      "defaultValue": null,
      "description": "Used in combination with dismissiable property to close the notification panel.",
      "name": "onClick",
      "parent": {
        "fileName": "src/components/NotificationPanel/NotificationPanel.tsx",
        "name": "NotificationPanelProps"
      },
      "declarations": [
        {
          "fileName": "src/components/NotificationPanel/NotificationPanel.tsx",
          "name": "NotificationPanelProps"
        }
      ],
      "required": false,
      "type": {
        "name": "((e?: MouseEvent<HTMLElement, MouseEvent>) => void) | undefined"
      }
    },
    "dismissable": {
      "defaultValue": null,
      "description": "Toggles the close button in the top right corner.",
      "name": "dismissable",
      "parent": {
        "fileName": "src/components/NotificationPanel/NotificationPanel.tsx",
        "name": "NotificationPanelProps"
      },
      "declarations": [
        {
          "fileName": "src/components/NotificationPanel/NotificationPanel.tsx",
          "name": "NotificationPanelProps"
        }
      ],
      "required": false,
      "type": {
        "name": "boolean"
      }
    },
    "fluid": {
      "defaultValue": null,
      "description": "Enables a fluid outer container that spans the entire width of parent.",
      "name": "fluid",
      "parent": {
        "fileName": "src/components/NotificationPanel/NotificationPanel.tsx",
        "name": "NotificationPanelProps"
      },
      "declarations": [
        {
          "fileName": "src/components/NotificationPanel/NotificationPanel.tsx",
          "name": "NotificationPanelProps"
        }
      ],
      "required": false,
      "type": {
        "name": "boolean"
      }
    },
    "label": {
      "defaultValue": null,
      "description": "Sets a label for the notification panel.",
      "name": "label",
      "parent": {
        "fileName": "src/components/NotificationPanel/NotificationPanel.tsx",
        "name": "NotificationPanelProps"
      },
      "declarations": [
        {
          "fileName": "src/components/NotificationPanel/NotificationPanel.tsx",
          "name": "NotificationPanelProps"
        }
      ],
      "required": false,
      "type": {
        "name": "string"
      }
    }
  }
}
