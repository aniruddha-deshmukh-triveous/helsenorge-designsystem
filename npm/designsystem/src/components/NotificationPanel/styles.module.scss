@import '../../scss/_spacers.scss';
@import '../../scss/_breakpoints.scss';
@import '../../scss/_font-settings.scss';
@import '../../scss/_palette.scss';
@import '../AnchorLink/styles.module.scss';

.notification-panel {
  display: grid;
  width: inherit;
  padding: $spacer;
  grid-template-areas: 'icon' 'content' 'action';
  grid-template-columns: 48px auto 48px;
  grid-template-rows: 1fr;

  &--small {
    max-width: map-get($grid-breakpoints, sm);
  }
  &--medium {
    max-width: map-get($grid-breakpoints, md);
  }
  &--large {
    max-width: map-get($grid-breakpoints, lg);
  }
  &--info {
    background-color: $kiwi50;
  }
  &--warn {
    background-color: $banana50;
  }
  &--alert {
    background-color: $cherry100;
  }
  &--crisis {
    background-color: $black;
  }
  &--shadow {
    box-shadow: 0 0.3rem getSpacer(xs) -0.1rem rgba(0, 0, 0, 0.15);
  }
  &--haslabel {
    padding: getSpacer(l) $spacer;
  }

  &__fluidwrapper {
    display: flex;
    justify-content: center;
    width: 100%;

    &--info {
      background-color: $kiwi50;
    }
    &--warn {
      background-color: $banana50;
    }
    &--alert {
      background-color: $cherry100;
    }
    &--crisis {
      background-color: $black;
    }
    &--shadow {
      box-shadow: 0 0.3rem getSpacer(xs) -0.1rem rgba(0, 0, 0, 0.15);
    }
  }

  &__icon {
    grid-area: 'icon';
  }

  &__content {
    grid-area: 'content';
    margin: getSpacer(xs) getSpacer(m);

    p {
      margin: 0;
    }

    a {
      @include anchorlink;
      @include anchorlink-icon;
    }

    &--crisis {
      color: $banana200;
      a {
        color: $white;
        &:hover {
          background-color: rgba(255, 255, 255, 0.2);
          color: $white;
        }
        &:focus {
          background-color: $white;
          color: $black;
          border-color: $white;
        }
        &[target='_blank'] {
          &:after {
            content: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' width='38px' height='38px' viewBox='0 0 48 48' fill='%23bdbab9'><polygon points='33.305,14.659,33.305,29.205,32.005,29.205,32.005,16.948,15.804,33.149,14.885,32.229,31.155,15.959,18.761,15.96,18.761,14.659' /></svg>");
          }

          &:focus {
            &:after {
              content: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg'  width='38px' height='38px' viewBox='0 0 48 48'><polygon points='33.305,14.659,33.305,29.205,32.005,29.205,32.005,16.948,15.804,33.149,14.885,32.229,31.155,15.959,18.761,15.96,18.761,14.659' /></svg>");
            }
          }
        }
      }
    }

    &--isred {
      color: $cherry500;
    }

    &--haslabel {
      display: flex;
      align-items: center;
      margin: 0 1rem;
    }
  }

  &__label {
    font-size: $font-size-sm;
    line-height: getSpacer(m);
    font-weight: 600;
    margin: 0;
    @media (min-width: map-get($grid-breakpoints, md)) {
      font-size: $font-size-md;
      line-height: $lineheight-size-md;
    }
  }

  &__action-column {
    grid-area: 'action';
  }

  &__close-button {
    display: flex;
    position: relative;
    background: transparent;
    border: 2px solid $blueberry600;
    padding: 0;
    justify-content: center;
    outline: none;
    cursor: pointer;
    &:before {
      content: ' ';
      position: absolute;
      top: -getSpacer(2xs);
      left: -getSpacer(2xs);
      right: -getSpacer(2xs);
      bottom: -getSpacer(2xs);
      border: 2px solid transparent;
    }
    &:after {
      content: ' ';
      position: absolute;
      top: -0.375rem;
      left: -0.375rem;
      right: -0.375rem;
      bottom: -0.375rem;
      border: 2px solid transparent;
    }
    &:hover,
    :active,
    :focus {
      background-color: $blueberry700;
      border-color: transparent;
      &:before {
        border-color: $blueberry700;
      }
      &:after {
        border-color: $blueberry700;
      }
    }
  }
}
