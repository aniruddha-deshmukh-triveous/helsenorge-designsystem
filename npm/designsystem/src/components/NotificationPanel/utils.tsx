import React, { AriaAttributes } from 'react';
import { palette } from '../../theme/palette';
import Icon from '../Icons';

import InfoSignStroke from '../Icons/InfoSignStroke';
import AlertSignStroke from '../Icons/AlertSignStroke';
import AlertSignFill from '../Icons/AlertSignFill';

export const variantToIconMap = {
  info: <Icon svgIcon={InfoSignStroke} color={palette.kiwi900} hoverColor={palette.kiwi900} />,
  warn: <Icon svgIcon={AlertSignStroke} color="black" hoverColor="black" />,
  alert: <Icon svgIcon={AlertSignFill} color={palette.cherry500} hoverColor={palette.cherry500} />,
  alertLabel: <Icon svgIcon={AlertSignStroke} color={palette.cherry500} hoverColor={palette.cherry500} />,
  crisis: <Icon svgIcon={AlertSignFill} color={palette.banana200} hoverColor={palette.banana200} />,
};

interface AriaLabelAttributesConfig {
  label?: string;
  id?: string;
}

type AriaLabelAttributes = Pick<AriaAttributes, 'aria-label' | 'aria-labelledby'>;

export const getAriaLabelAttributes = (config: AriaLabelAttributesConfig): AriaLabelAttributes | undefined => {
  const { label, id } = config;

  if (id) {
    return {
      'aria-labelledby': id,
    };
  }
  if (label) {
    return {
      'aria-label': label,
    };
  }
};
