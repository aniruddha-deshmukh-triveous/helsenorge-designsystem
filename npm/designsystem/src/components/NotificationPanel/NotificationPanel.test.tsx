import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import NotificationPanel from './NotificationPanel';

describe('Gitt at NotificationPanel skal vises', () => {
  describe('Når panelet vises med default props', () => {
    test('Så rendres komponenten riktig', (): void => {
      const { container } = render(<NotificationPanel>Some text here for testing.</NotificationPanel>);
      expect(container).toMatchSnapshot();
    });
  });
  describe('Når panelet kan lukkes', () => {
    test('Så kalles onClick-handleren når man klikker på knappen', (): void => {
      const mockOnClick = jest.fn();
      const { container } = render(
        <NotificationPanel dismissable onClick={mockOnClick}>
          Some text here for testing.
        </NotificationPanel>
      );
      expect(container).toMatchSnapshot();

      const closeButton = screen.getByRole('button');
      userEvent.click(closeButton);

      expect(mockOnClick).toHaveBeenCalledTimes(1);
    });
  });
});
