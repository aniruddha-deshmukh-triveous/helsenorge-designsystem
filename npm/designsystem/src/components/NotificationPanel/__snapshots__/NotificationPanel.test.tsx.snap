// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gitt at NotificationPanel skal vises Når panelet kan lukkes Så kalles onClick-handleren når man klikker på knappen 1`] = `
<div>
  <div
    class="notification-panel notification-panel--info notification-panel--has-children notification-panel--dismissable"
    data-analyticsid="notification-panel"
  >
    <span
      class="notification-panel__icon"
    >
      <svg
        aria-hidden="true"
        class="hnds-style-icon "
        data-analyticsid="icon"
        fill="#078141"
        height="48"
        role="img"
        style="min-width: 48px; min-height: 48px;"
        viewBox="0 0 48 48"
        width="48"
      >
        <path
          d="M25.672 30.501a.682.682 0 01-.322-.272.835.835 0 01-.11-.443v-7.917l-.112-.112-3.651.194v.544c.148.015.314.05.495.104.18.055.316.112.406.172a.994.994 0 01.299.316.88.88 0 01.119.459v6.337a.879.879 0 01-.098.443.623.623 0 01-.335.257c-.09.035-.197.06-.32.074a8.883 8.883 0 01-.38.038v.544h4.709v-.544a2.45 2.45 0 01-.7-.194M24 37.604c-7.5 0-13.604-6.103-13.604-13.604 0-7.501 6.104-13.603 13.604-13.603 7.502 0 13.602 6.102 13.602 13.603 0 7.501-6.1 13.604-13.601 13.604m0-28.909C15.56 8.695 8.694 15.562 8.694 24c0 8.438 6.867 15.305 15.305 15.305 8.438 0 15.304-6.867 15.304-15.305 0-8.438-6.866-15.305-15.304-15.305m-.274 10.632a1.559 1.559 0 100-3.118 1.559 1.559 0 000 3.118"
        />
      </svg>
    </span>
    <section
      aria-label="Some text here for testing."
      class="notification-panel__content"
    >
      Some text here for testing.
    </section>
    <span
      class="notification-panel__close"
    >
      <button
        aria-label="Lukk"
        class="close"
        data-analyticsid="close"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="hnds-style-icon "
          data-analyticsid="icon"
          fill="#126F87"
          height="48"
          role="img"
          style="min-width: 48px; min-height: 48px;"
          viewBox="0 0 48 48"
          width="48"
        >
          <path
            d="M25.403 24l10.259-10.259-1.403-1.403L24 22.597l-10.259-10.26-1.403 1.403L22.597 24 12.338 34.26l1.403 1.403L24 25.403l10.259 10.259 1.403-1.403z"
          />
        </svg>
      </button>
    </span>
  </div>
</div>
`;

exports[`Gitt at NotificationPanel skal vises Når panelet vises med default props Så rendres komponenten riktig 1`] = `
<div>
  <div
    class="notification-panel notification-panel--info notification-panel--has-children"
    data-analyticsid="notification-panel"
  >
    <span
      class="notification-panel__icon"
    >
      <svg
        aria-hidden="true"
        class="hnds-style-icon "
        data-analyticsid="icon"
        fill="#078141"
        height="48"
        role="img"
        style="min-width: 48px; min-height: 48px;"
        viewBox="0 0 48 48"
        width="48"
      >
        <path
          d="M25.672 30.501a.682.682 0 01-.322-.272.835.835 0 01-.11-.443v-7.917l-.112-.112-3.651.194v.544c.148.015.314.05.495.104.18.055.316.112.406.172a.994.994 0 01.299.316.88.88 0 01.119.459v6.337a.879.879 0 01-.098.443.623.623 0 01-.335.257c-.09.035-.197.06-.32.074a8.883 8.883 0 01-.38.038v.544h4.709v-.544a2.45 2.45 0 01-.7-.194M24 37.604c-7.5 0-13.604-6.103-13.604-13.604 0-7.501 6.104-13.603 13.604-13.603 7.502 0 13.602 6.102 13.602 13.603 0 7.501-6.1 13.604-13.601 13.604m0-28.909C15.56 8.695 8.694 15.562 8.694 24c0 8.438 6.867 15.305 15.305 15.305 8.438 0 15.304-6.867 15.304-15.305 0-8.438-6.866-15.305-15.304-15.305m-.274 10.632a1.559 1.559 0 100-3.118 1.559 1.559 0 000 3.118"
        />
      </svg>
    </span>
    <section
      aria-label="Some text here for testing."
      class="notification-panel__content"
    >
      Some text here for testing.
    </section>
  </div>
</div>
`;
